### ISSUE TITLE
Trace of torch.tensor is impressively convoluted

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#29063]
```
    def test_trace_modern_ctor(self):
        class MyModule(nn.Module):
            def forward(self, x):
                return torch.tensor([0.0])

        # Shouldn't fail sanity checks
        traced_rec = torch.jit.trace(MyModule(), torch.randn(2, 2))
        print(traced_rec.graph)
```

gives the trace

```
graph(%self : ClassType<Module>,                                                                                          
      %1 : Double(2, 2)):
  %2 : Double(1) = prim::Constant[value={0}](), scope: MyModule # test/test_jit.py:321:0
  %3 : Device = prim::Constant[value="cpu"](), scope: MyModule # test/test_jit.py:321:0
  %4 : int = prim::Constant[value=7](), scope: MyModule # test/test_jit.py:321:0
  %5 : bool = prim::Constant[value=0](), scope: MyModule # test/test_jit.py:321:0
  %6 : bool = prim::Constant[value=0](), scope: MyModule # test/test_jit.py:321:0
  %7 : None = prim::Constant(), scope: MyModule
  %8 : Double(1) = aten::to(%2, %3, %4, %5, %6, %7), scope: MyModule # test/test_jit.py:321:0
  %9 : Double(1) = aten::detach(%8), scope: MyModule # test/test_jit.py:321:0
  return (%9)
```

This is way more nodes than I expected to see.

cc @suo

**Comment by suo** [comment]
@ezyang I remember there was some conversation on chat, but did we decide that the optimized graph was "good enough"?

**Comment by ezyang** [comment]
@zdevito said:

> The detach call is weird, yes.
> But

```
     argument_1: Tensor) -> Tensor:
  _0 = torch.to(CONSTANTS.c0, torch.device("cpu"), 6, False, False, None)
  return torch.detach(_0)
```

> Is what I would expect except for the detach call
> I expect one call to some constructor object but the rough number of nodes seems fine, since mosts are constants. This is the unoptimized graph though

```
      %1 : Float(*, *)):
  %11 : Float(1) = prim::Constant[value={0}]()
  return (%11)
```

> That is the optimized graph. it looks bad unoptimized but once it is optimized it turns into a single tensor constant as you would expect.

@eellison added:

> For any non-trivial trace all of those constants will likely appear anyway, and be constant pooled at the beginning of the graph. Agreed this is a lot for one op tho

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
